distribution/tools/entitlement-runtime/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyBuilder.java

```diff
@@ -0,0 +1,110 @@
+                );
+            }
+            if (policyParser.nextToken() != XContentParser.Token.FIELD_NAME
+                || policyParser.currentName().equals(POLICY_PARSEFIELD.getPreferredName()) == false) {
+                throw newPolicyParserException(
+                    policyParser.getTokenLocation(),
+                    policyName,
+                    "expected object [" + POLICY_PARSEFIELD.getPreferredName() + "]"
+                );
+            }
+            if (policyParser.nextToken() != XContentParser.Token.START_ARRAY) {
+                throw newPolicyParserException(
+                    policyParser.getTokenLocation(),
+                    policyName,
+                    "expected array of [" + ModuleScopeBuilder.MODULE_PARSEFIELD.getPreferredName() + "]"
+                );
+            }
+            while (policyParser.nextToken() != XContentParser.Token.END_ARRAY) {
+                if (policyParser.currentToken() != XContentParser.Token.START_OBJECT) {
+                    throw newPolicyParserException(
+                        policyParser.getTokenLocation(),
+                        policyName,
+                        "expected object [" + ModuleScopeBuilder.MODULE_PARSEFIELD.getPreferredName() + "]"
+                    );
+                }
+                if (policyParser.nextToken() != XContentParser.Token.FIELD_NAME
+                    || policyParser.currentName().equals(ModuleScopeBuilder.MODULE_PARSEFIELD.getPreferredName()) == false) {
+                    throw newPolicyParserException(
+                        policyParser.getTokenLocation(),
+                        policyName,
+                        "expected object [" + ModuleScopeBuilder.MODULE_PARSEFIELD.getPreferredName() + "]"
+                    );
+                }
+                if (policyParser.nextToken() != XContentParser.Token.START_OBJECT) {
+                    throw newPolicyParserException(
+                        policyParser.getTokenLocation(),
+                        policyName,
+                        "expected object [" + ModuleScopeBuilder.MODULE_PARSEFIELD.getPreferredName() + "]"
+                    );
+                }
+                ModuleScopeBuilder moduleScopeBuilder = new ModuleScopeBuilder(policyName, policyParser);
+                moduleScopeBuilder.parseScope();
+                policyModules.add(moduleScopeBuilder);
+                if (policyParser.nextToken() != XContentParser.Token.END_OBJECT) {
+                    throw newPolicyParserException(policyParser.getTokenLocation(), policyName, "expected closing object");
+                }
+            }
+            if (policyParser.nextToken() != XContentParser.Token.END_OBJECT) {
+                throw newPolicyParserException(policyParser.getTokenLocation(), policyName, "expected closing object");
+            }
+        } catch (IOException ioe) {
+            throw new UncheckedIOException(ioe);
```

Comment:
Given that this is using an `InputStream`, should the method be declared `throws IOException`?

Reply:
Changed.